Assets/LeapMotionModules/UI/LeapMotionHandsExtentionAPI/Hands.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Leap;
+using Leap.Unity;
+using UnityEngine;
+using System.Collections.Generic;
+
+namespace Leap.Unity {
+
+  public static class Hands {
+
+    private static LeapServiceProvider s_provider;
+
+    static Hands() {
+      s_provider = GameObject.FindObjectOfType<LeapServiceProvider>();
```

Comment:
This logic will fail under any case where the provider is destroyed and recreated.  The variable s_provider will still reference the old destroyed provider, with nothing able to cause it to be re-initialized to point to the new provider.  Probably the most common case of this would be when switching scenes.